src/main/java/com/google/devtools/build/lib/remote/blobstore/http/HttpBlobStore.java

```diff
@@ -105,8 +116,43 @@
   @GuardedBy("credentialsLock")
   private long lastRefreshTime;
 
-  @SuppressWarnings("FutureReturnValueIgnored")
-  public HttpBlobStore(
+  public static HttpBlobStore create(URI uri, int timeoutMillis,
+      int remoteMaxConnections, @Nullable final Credentials creds)
+      throws Exception {
+    return new HttpBlobStore(
+        new InetSocketAddress(uri.getHost(), uri.getPort()),
+        NioEventLoopGroup::new,
+        NioSocketChannel.class,
+        uri, timeoutMillis, remoteMaxConnections, creds);
+  }
+
+  public static HttpBlobStore create(
```

Comment:
Please just plain old constructor overloads here.

Reply:
I don't think it's possible to use simple constructors for this without introducing an additional class to tuple the `newEventLoopGroup` and `channelClass` parameters together.

Would it be best to create a new `HttpBlobStoreFactory` and move these create methods there?